common/SolidityTestUtils.js

```diff
@@ -115,26 +115,34 @@ async function revertToSnapshot(web3, id) {
 async function mineTransactionsAtTime(web3, transactions, time, sender) {
   await stopMining(web3);
 
-  const receiptPromises = [];
-  for (const transaction of transactions) {
-    const result = transaction.send({ from: sender });
+  try {
+    const receiptPromises = [];
+    for (const transaction of transactions) {
+      const result = transaction.send({ from: sender });
 
-    // Awaits the transactionHash, which signifies the transaction was sent, but not necessarily mined.
-    await new Promise((resolve, reject) => {
-      result.on("transactionHash", function() {
-        resolve();
+      // Awaits the transactionHash, which signifies the transaction was sent, but not necessarily mined.
+      await new Promise((resolve, reject) => {
+        result.on("transactionHash", function() {
+          resolve();
+        });
+        result.on("error", function(error) {
```

Comment:
@mrice32 while trying to use `mineTransactionsAtTime()` to unit test this PR, I realized that this method should better catch reverted transactions  so that it won't exit the program while keeping the ganache miner stopped.